modules/ROOT/edits/execution-engine.adoc

```diff
@@ -0,0 +1,273 @@
+# Supports Expressions
+# Only applies when org.mule.runtime.scheduler.threadPool.strategy=DEDICATED
+org.mule.runtime.scheduler.io.threadPool.coreSize=cores
+
+# The maximum number of threads to allow in the I/O pool.
+# Supports Expressions
+# Only applies when org.mule.runtime.scheduler.threadPool.strategy=DEDICATED
+org.mule.runtime.scheduler.io.threadPool.maxSize=max(2, cores + ((mem - 245760) / 5120))
+
+# The size of the queue to use for holding I/O tasks before they are executed.
+# Supports Expressions
+# Only applies when org.mule.runtime.scheduler.threadPool.strategy=DEDICATED
+org.mule.runtime.scheduler.io.workQueue.size=0
+
+# When the number of threads in the I/O pool is greater than SchedulerService.io.coreThreadPoolSize, this is the maximum
+# time (in milliseconds) that excess idle threads will wait for new tasks before terminating.
+# Only applies when org.mule.runtime.scheduler.threadPool.strategy=DEDICATED
+org.mule.runtime.scheduler.io.threadPool.threadKeepAlive=30000
+
+# The number of threads to keep in the cpu_intensive pool, even if they are idle.
+# Supports Expressions
+# Only applies when org.mule.runtime.scheduler.threadPool.strategy=DEDICATED
+org.mule.runtime.scheduler.cpuIntensive.threadPool.size=2*cores
+
+# The size of the queue to use for holding cpu_intensive tasks before they are executed.
+# Supports Expressions
+# Only applies when org.mule.runtime.scheduler.threadPool.strategy=DEDICATED
+org.mule.runtime.scheduler.cpuIntensive.workQueue.size=2*cores
+----
+
+==== Considerations
+
+MuleSoft doesn’t recommend changing the used pool strategy nor change its configuration values. Note that the configuration is global and affects the entire Mule Runtime instance.
+
+MuleSoft recommends that you perform load and stress testing with all applications involved in real-life scenarios to validate any change in the threading configurations and to understand how the thread pools work in Mule 4.
+
+=== Configuration at the Application Level
+
+You can define the pool strategy to use in an application by adding the following to your application code:
+
+[source, xml, linenums]
+----
+<ee:scheduler-pools poolStrategy="UBER" gracefulShutdownTimeout="15000">
+   <ee:uber
+       corePoolSize="1"
+       maxPoolSize="9"
+       queueSize="5"
+       keepAlive="5"/>
+</ee:scheduler-pools>
+----
+
+Notice that there’s a `poolStrategy` parameter. This parameter exists for backwards compatibility reasons and allows you to switch back to the three pools scheme from prior versions of mule:
+
+[source, xml, linenums]
+----
+<ee:scheduler-pools gracefulShutdownTimeout="15000">
+   <ee:cpu-light
+           poolSize="2"
+           queueSize="1024"/>
+   <ee:io
+           corePoolSize="1"
+           maxPoolSize="2"
+           queueSize="0"
+           keepAlive="30000"/>
+   <ee:cpu-intensive
+           poolSize="4"
+           queueSize="2048"/>
+</ee:scheduler-pools>
+----
+
+==== Considerations
```

Comment:
ditto re. removing solitary Considerations sections and merging them into their containers.

Reply:
Even though the following text is constructed differently from in the previous PR, I think that it can be reorganized and reworded to match it more closely. Suggest this order:
1. Mule recommends the default.
2. But you can customize.'
3. If you do, consult Support before production.

Reply:
The existing org is copyedited only next.